src/utils/io/random_access.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use positioned_io::{ReadAt, Size};
+use std::fs::File;
+use std::io;
+use std::path::Path;
+
+/// This wrapper is needed, because [`positioned_io::RandomAccessFile`] does not support
+/// [`positioned_io::Size`].
+pub struct RandomAccessFile {
+    file: positioned_io::RandomAccessFile,
+    size: u64,
+}
+
+impl RandomAccessFile {
+    /// This opens a file and fetches the size. Fails with an error if the path is a directory.
+    pub fn open<P: AsRef<Path>>(path: P) -> io::Result<RandomAccessFile> {
+        let file = File::open(path)?;
+        let size = file.metadata()?.len();
+        let file = positioned_io::RandomAccessFile::try_new(file)?;
+        Ok(Self { file, size })
+    }
+}
+
+impl Size for RandomAccessFile {
+    fn size(&self) -> std::io::Result<Option<u64>> {
+        Ok(Some(self.size))
```

Comment:
I think this should mirror the implementation for `File`
https://docs.rs/positioned-io/latest/src/positioned_io/lib.rs.html#333-342

Reply:
We don't have access to file as it's a private field in RandomAccessFile.

Reply:
There is no need to mirror that impl, because we throw the error already in the constructor if the `File` is a `dir`.

Reply:
However, I can use the `size` method in the constructor.